### ISSUE TITLE
ExternalRefProcessor incorrectly handles composed schemas in array items

### ISSUE THREAD

**Issue by chimmi** [swagger-api/swagger-parser#2071]
If we have following spec:
```
openapi: 3.0.1
info:
  title: API
  description: API
  version: LATEST
paths:
  /test-path:
    post:
      requestBody:
        required: true
        content:
          text/plain:
            schema:
              type: string
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                $ref: 'definitions.yaml#/components/schemas/Response'
```
And corresponding `definitions.yaml`:
```
components:
  schemas:
    Response:
      type: array
      items:
        oneOf:
          - $ref: '#/components/schemas/ProductRow'
        discriminator:
          propertyName: type
          mapping:
            'product': '#/components/schemas/ProductRow'
    ProductRow:
      type: object
      additionalProperties: false
      required:
        - type
      properties:
        type:
          $ref: '#/components/schemas/ProductsRowType'
        payload:
          type: string
    ProductsRowType:
      type: string
      enum:
        - product
```
`ExternalRefProcessor` will fail to resolve all necessary references, which results in only `Response` model being loaded.

**Comment by chimmi** [comment]
[PR] (https://github.com/swagger-api/swagger-parser/pull/2072)

**Comment by frantuma** [comment]
fixed in #2072, thanks!
